libultraship/libultraship/Resource.h

```diff
@@ -101,10 +101,10 @@ namespace Ship
 
     class ResourcePromise {
     public:
-        std::shared_ptr<Resource> Resource;
-        std::shared_ptr<File> File;
-        std::condition_variable ResourceLoadNotifier;
-        std::mutex ResourceLoadMutex;
+        std::shared_ptr<Resource> resource;
```

Comment:
Copy paste my above comments about refactoring for the sake of it versus if it is necessary for this PR. It's probably best to be careful especially because these are the files that have the segfault in them. (But these are better variable names to be sure)

Reply:
Regarding the member reordering, refer to my previous reply. As for the member renaming, these identifiers actually cause conflicts and make GCC output a warning.
I don't remember what the standard says exactly about it though.
IIRC clang and MSVC both don't warn about it.